src/main.cpp

```diff
@@ -3040,6 +3056,41 @@ static bool ShouldCheckTransactions(const CChainParams& chainparams, const CBloc
              && Checkpoints::IsAncestorOfLastCheckpoint(chainparams.Checkpoints(), pindex));
 }
 
+template<typename Tree>
+bool SubtreeStateConsistent(
+    const ShieldedType type,
+    const Tree& tree,
+    const CCoinsView* view
+)
+{
+    auto latestSubtree = view->GetLatestSubtree(type);
+
+    if (latestSubtree) {
+        // The view believes the last completed subtree is
+        // at index latestSubtree->index, so the "current"
+        // subtree index of the tree should be one
+        // larger than this.
+        if ((latestSubtree->index + 1) == tree.current_subtree_index()) {
```

Comment:
Another example of where having an accessor for `index + 1` (i.e. the index of the next subtree) would help to reduce explicit `+ 1`s.

Reply:
This was far simpler, thanks.